### ISSUE TITLE
Spanish Translation

### ISSUE THREAD

**Issue by fersdt** [learnlatex/learnlatex.github.io#66]
Hi! 
   I would be pleased to help you with the Spanish translation of your site if you are interested in. I'm a Spaniard who live in France where I work as Physics and Chemistry teacher.
The contents of you site are great!!

**Comment by davidcarlisle** [comment]
Yes we would certainly be interested in a translation.

We have had another informal offer

https://chat.stackexchange.com/transcript/message/55052473#55052473

I have asked that stackexchange user to comment here, perhaps the two of you can collaborate, having an extra pair of eyes viewing the text is always useful!

Just one warning about translations, while we have finished the main work of producing the English text so it is "safe" to start translating, the course has not really been used by real learners yet and we may want to adjust some things after user experience or in response to github issues, so there may be periodic requests to update the text based on the diff in the English version from whichever point you fork the translation.

**Comment by fersdt** [comment]
That's great :-) . Four eyes are better than two

**Comment by EmanuelPrima** [comment]
Hi @fersdt ! I am also interested in translating the site to Spanish. We can make a team.

In addition to @davidcarlisle's comment, you can start reading https://github.com/learnlatex/learnlatex.github.io/blob/master/TRANSLATIONS.md for further information if you haven't done it before. I still have to review that document in detail, but if you want you can start with the translation that I will join later.

**Comment by fersdt** [comment]
Hi @EmanuelPrima ! 
      I have follow the instructions given in the TRANSLATIONS.md file. We can work in the following repository : 
https://github.com/fersdt/learnlatex.github.io
      I have just sent you an invitation to this repository.
Un placer poder trabajar con un compañero :-)

**Comment by fersdt** [comment]
Hi @EmanuelPrima!
   I have just added a file named "Translations_ES" in the main folder of our folk with the translated files, you can update it freely. That will allow us to see the evolution during the translation process.
   We can also add a mark, for exemple "(ok)", for each item in this file when a translation have been checked by one of us.
   Feel free to change the translations if you think something is wrong.

**Comment by EmanuelPrima** [comment]
Excellent @fersdt!

**Comment by EmanuelPrima** [comment]
@fersdt I was looking your translation and I feel quite comfortable :)

However, I read some Q&A that suggets the use of angular quotation marks for citing text as the first symbols we should use. The list that is proposed based on [this](https://www.rae.es/dpd/comillas) and [this](https://spanish.stackexchange.com/a/939/21271) is:

1. Angular quotation marks (« »).
2. Double quotation marks (" ").
3. Single quotation marks (' ').

I don't think using the first would be a good option for the purposes of the site, but double quotation marks are the most used. As examples look at this newspaper news: [Coronavirus: las aplicaciones más inclusivas de la cuarentena](https://www.infobae.com/america/tecno/2020/07/29/coronavirus-las-aplicaciones-mas-inclusivas-de-la-cuarentena/) and [Barcelona: Arthur, "el petardazo final" en medio de la crisis del club catalán](https://www.lanacion.com.ar/deportes/futbol/barcelona-arthur-el-petardazo-final-medio-crisis-nid2407444).

What do you think?

**Comment by fersdt** [comment]
@EmanuelPrima 
I completely agree with you. I have used the same strategy. I have always used the double quotation marks. :-)
But the text is written in markdown. Some times in the original text, the backtick is used (for exemple `pdflatex`). In markdown, that allows to highlight a word (usually as a code word).
You can see a list of marks used in markdown here
https://www.markdownguide.org/basic-syntax/

**Comment by fersdt** [comment]
Sorry @EmanuelPrima 
I realise that even if I have used the double quote in the translated text, it's the single quotation that appears in the site.
I'm going to fix it.

**Comment by fersdt** [comment]
@EmanuelPrima Yo are right, I made some mistakes in index.md and lesson-01.md where I used some single quotes. The problem should now fixed. Four eyes are better than two ;-)

**Comment by fersdt** [comment]
Hi @davidcarlisle,

   During the Spanish translation of the file **more-07.md** I found a minor error in a link (in line: *97*) `https://ctan.org/pkg/trivloat` should be `https://ctan.org/pkg/trivfloat`. I fixed this bug in the Spanish file. Probably you can fix it in the other languages.

**Comment by davidcarlisle** [comment]
@fersdt  thanks.

**Comment by EmanuelPrima** [comment]
@fersdt @davidcarlisle  In `lesson-03.md` and following, the spanish translations are applied to the examples. This includes the use of some special characters such as accents, `¡`, `ñ` etc.

David, should we add this 3 lines from that lesson on??:

```latex
...
\usepackage[english]{babel}
\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
...
```

If the answer is 'yes', I don't understand [Language specific lessons](https://github.com/learnlatex/learnlatex.github.io/blob/master/TRANSLATIONS.md#translating-examples) so can you help us by telling us what to do, please?

**Comment by joulev** [comment]
@EmanuelPrima No, please don’t translate examples in ‘normal’ lessons. Leave them in English, only translate the comments. The language specific lessons are for “how I can typeset this language in LaTeX”, and that is where you add your own Spanish examples.

**Comment by davidcarlisle** [comment]
@EmanuelPrima we wrote the instructions before anyone tried doing this so it may be that we need to change things as we get more experience but...

Two styles of translation are described in https://www.learnlatex.org/TRANSLATIONS, one in which the examples are kept unchanged (this style is used by the Vietnamese and Portuguese translations already checked in) and one in which examples are translated (at least two of the translations currently being worked on are in this style, I believe).

See 
https://www.learnlatex.org/TRANSLATIONS#translating-examples
for comments on how we thought that the translated examples could be arranged.

Basically you can _describe_ the Spanish babel setup (and any other specific points about Spanish use) in the language-xx lessons but just have a forward reference there  from the first example in lesson-03 and from the help file.
So just add a sentence under the example saying the extra lines are for Spanish language setup described in language-01.

The Japanese help file is currently faked (by me and google translate) but in

https://www.learnlatex.org/ja/help

I sketch how you might have two examples there showing both the original English example from en/help and a translated one.
There is a bigger difference there as one is using pdflatex and the other the Japanese platex engine, but the same basic idea could be used  here.

**Comment by davidcarlisle** [comment]
@EmanuelPrima  also please don't add

```
\usepackage[utf8]{inputenc}
```

That is the default in latex and one reason for ensuring that the back end servers used for examples were up to date is to endure that examples would not need that line.

**Comment by fersdt** [comment]
@davidcarlisle I miss problably something but in Portuguese, the examples are translated. 
I think that's more useful for better comprehension of the lessons.
What do yo think @EmanuelPrima ? We can continue to follow the second strategy. Examples are translated from lesson 1 to lesson 7.

**Comment by davidcarlisle** [comment]
@fersdt  sorry I mis-remembered, I will edit the above comment so it is true re Portuguese.

**Comment by PhelypeOleinik** [comment]
@fersdt I did translate the examples to Portuguese because other than the hyphenation patterns, the setup for English works quite well for Portuguese, so other than “Hello World” _vs_ “Olá Mundo” there isn't any difference in what the example shows, so other than the language of the text there's no change in content... But I may as well revert that. @davidcarlisle Should I?

**Comment by davidcarlisle** [comment]
@PhelypeOleinik Actually I think the examples should be "good practice" (people will copy them, whatever the surrounding text says:-) so I think probably if they are translated they should have babel/fontenc calls to set up correct hyphenation.

Being a native English speaker I really am  trying to avoid recommending whether you translate the examples or not.
In some communities the main use of latex is probably technical articles in English language journals, so English examples with translated lesson texts makes perfect sense. In other communities that will look less natural.

It would also be possible to translate the initial "hello world" examples but then switch to English examples for more technical lessons like math. I think as long as you describe in the help page what you have done, we need to allow some flexibility here.

**Comment by davidcarlisle** [comment]
@josephwright  I do wonder if we should use 

`\usepackage[T1]{fontenc}`

as part of the standard boilerplate in English examples, it would minimize differences in translations and is probably good practice anyway

**Comment by josephwright** [comment]
@davidcarlisle I suspect we should: will need to have an explanation saying 'see about packages' or something

**Comment by fersdt** [comment]
Hi !

     I have finish the first version of the Spanish Translation. I'm going to reread today the whole translation in order to avoid typo and other minor errors.
     We can probably merge this (the es directory) tomorrow. What do you think about @EmanuelPrima ? Other minor errors could be fixed later in the main branch.

**Comment by davidcarlisle** [comment]
Thank you  for working on learnlatex.org translations.
You may have seen already if watching the repository but just a note
to say that we have extended lesson-06 and more-06 to add descriptions
of defining commands with \newcommand and \NewDocumentCommand

https://github.com/learnlatex/learnlatex.github.io/commit/463d181301d667b10bf23c164f9d5ae15371877e

**Comment by pablgonz** [comment]
> @fersdt I was looking your translation and I feel quite comfortable :)
> 
> However, I read some Q&A that suggets the use of angular quotation marks for citing text as the first symbols we should use. The list that is proposed based on [this](https://www.rae.es/dpd/comillas) and [this](https://spanish.stackexchange.com/a/939/21271) is:
> 
>     1. Angular quotation marks (« »).
> 
>     2. Double quotation marks (" ").
> 
>     3. Single quotation marks (' ').
> 
> 
> I don't think using the first would be a good option for the purposes of the site, but double quotation marks are the most used. As examples look at this newspaper news: [Coronavirus: las aplicaciones más inclusivas de la cuarentena](https://www.infobae.com/america/tecno/2020/07/29/coronavirus-las-aplicaciones-mas-inclusivas-de-la-cuarentena/) and [Barcelona: Arthur, "el petardazo final" en medio de la crisis del club catalán](https://www.lanacion.com.ar/deportes/futbol/barcelona-arthur-el-petardazo-final-medio-crisis-nid2407444).
> 
> What do you think?

Hola a todos, solo un comentario respecto de esto (lo escribo en español primero), las comillas españolas (« ») son las correctas y las que se deberías usar, no es que las comillas dobles estén erradas, solo son comunes porque el teclado no trae de manera directa nuestras lindas comillas (y son parte del legado de la las maquinas de escribir de antaño).
Dejo aquí algunos enlaces bastante interesantes,  los primeros del genial maestro Javier Bezos (@jbezos), todo un referente en el tema,  y los otros asociados a las traducciones técnicas a nuestro idioma:

- http://www.texnia.com/archive/tipografia.pdf
- http://www.texnia.com/ortotipografia_estilo.html
- http://esteve.org/wp-content/uploads/2018/01/13231.pdf
- https://www.esteve.org/wp-content/uploads/2018/01/13226.pdf

Solo agradecer por el esfuerzo y la gran tarea que han desarrollado.

Ahora en Inglés (courtesy of deepl)

Hello everyone, just a comment about this (I write it in Spanish first), the Spanish quotation marks (« ») are the correct ones and the ones you should use, not that the double quotation marks are wrong, they are just common because the keyboard does not bring directly our nice quotation marks (and they are part of the legacy of the typewriters of yesteryear).
I leave here some interesting links, the first ones from the great master Javier Bezos (@jbezos), a reference on the subject, and the others associated with the technical translations to our language.

Just thanks for the effort and the great task they have developed

**Comment by jbezos** [comment]
Pablo, ¿adonde te envío el jamón? (the literal translation is ‘where do I send you the “jamón“,’ but it's a Spanish joke).

**Comment by davidcarlisle** [comment]
As you may have seen, the main site at learnlatex.org has been given a new design and some of the details of the language data have changed. 

If you are still working on the translation, I have updated
https://www.learnlatex.org/TRANSLATIONS
to hopefully give correct instructions for the new layout, but if anything is not clear please ask

**Comment by fersdt** [comment]
Hi David !
    Sorry for the delay. I have some time to translate the new details during this week.

**Comment by JairoAdelRio** [comment]
Hi everyone. I was told to ask/report here about typos and similar in the site. I found some typos and weird syntax, perhaps a calque from English versions. 

Ya en español, he encontrado algunos pequeños detalles, como aquí: https://www.learnlatex.org/es/lesson-06. Dice «Extendendiento» y algunos otros ejemplos requieren retoques. Estaré complacido de ayudar con el reporte de errores.

**Comment by fersdt** [comment]
No dudes en comentar los errores que encuentres. Es probable que haya algunos errores tipográficos como el que comentss. 👍

**Comment by fersdt** [comment]
> > @fersdt I was looking your translation and I feel quite comfortable :)
> > However, I read some Q&A that suggets the use of angular quotation marks for citing text as the first symbols we should use. The list that is proposed based on [this](https://www.rae.es/dpd/comillas) and [this](https://spanish.stackexchange.com/a/939/21271) is:
> > ```
> > 1. Angular quotation marks (« »).
> > 
> > 2. Double quotation marks (" ").
> > 
> > 3. Single quotation marks (' ').
> > ```
> > 
> > 
> > I don't think using the first would be a good option for the purposes of the site, but double quotation marks are the most used. As examples look at this newspaper news: [Coronavirus: las aplicaciones más inclusivas de la cuarentena](https://www.infobae.com/america/tecno/2020/07/29/coronavirus-las-aplicaciones-mas-inclusivas-de-la-cuarentena/) and [Barcelona: Arthur, "el petardazo final" en medio de la crisis del club catalán](https://www.lanacion.com.ar/deportes/futbol/barcelona-arthur-el-petardazo-final-medio-crisis-nid2407444).
> > What do you think?
> 
> Hola a todos, solo un comentario respecto de esto (lo escribo en español primero), las comillas españolas (« ») son las correctas y las que se deberías usar, no es que las comillas dobles estén erradas, solo son comunes porque el teclado no trae de manera directa nuestras lindas comillas (y son parte del legado de la las maquinas de escribir de antaño).
> Dejo aquí algunos enlaces bastante interesantes, los primeros del genial maestro Javier Bezos (@jbezos), todo un referente en el tema, y los otros asociados a las traducciones técnicas a nuestro idioma:
> 
>     * http://www.texnia.com/archive/tipografia.pdf
> 
>     * http://www.texnia.com/ortotipografia_estilo.html
> 
>     * http://esteve.org/wp-content/uploads/2018/01/13231.pdf
> 
>     * https://www.esteve.org/wp-content/uploads/2018/01/13226.pdf
> 
> 
> Solo agradecer por el esfuerzo y la gran tarea que han desarrollado.
> 
> Ahora en Inglés (courtesy of deepl)
> 
> Hello everyone, just a comment about this (I write it in Spanish first), the Spanish quotation marks (« ») are the correct ones and the ones you should use, not that the double quotation marks are wrong, they are just common because the keyboard does not bring directly our nice quotation marks (and they are part of the legacy of the typewriters of yesteryear).
> I leave here some interesting links, the first ones from the great master Javier Bezos (@jbezos), a reference on the subject, and the others associated with the technical translations to our language.
> 
> Just thanks for the effort and the great task they have developed

Muchas gracias por el comentario @pablgonz, es cierto que la traducción de un curso de LaTeX que no sigue la tipografía española como que no queda muy bien. He hecho los cambios a las comillas españolas y el texto queda mucho mejor. 
No dudes en hacer otros comentarios sobre mejoras de la traducción o errores que puedas ver. Somos humanos :-)

Tranks a lot for you comment @pablgonz, it's true that a Spanish Traduction of a LaTeX's course that doesn't respect the Spanish typography is awful. Spanish quotation marks are now used in the Trad and the text seems better. Do not hesitate to share with us any other comment to improve the Trad or to tell us the typos you can detect. We are only humans !

**Comment by pablgonz** [comment]
@fersdt Yes, I looked it up and they look great :)

**Comment by fersdt** [comment]
Hi,
    I have just realize a little problem, concerning the number of the lesson that appears in the top of a lesson's page. 
    Usually, this number is automatically update, but in the Spanish version of learnlatex.org, this number is always zero for the "main lessons" pages. However this number is ok for the "more on this topic" pages.
    I guess I missed something when translating but I don't know what.
Thanks in advance

**Comment by davidcarlisle** [comment]
@fersdt  thanks for the notice, I'll have a look...

**Comment by davidcarlisle** [comment]
@fersdt  ah. `es` is a substring of `lesson` .......

Sorry, I'll fix...

**Comment by davidcarlisle** [comment]
Fixed!

**Comment by fersdt** [comment]
Sorry for this message @davidcarlisle
I have just seen a new page AUTHORS.md
Is it possible to add me to this list ?
Thanks in advance.

**Comment by davidcarlisle** [comment]
Yes sure (you could have added yourself in the PR that added the translation:-) I'll do it now.

**Comment by pablgonz** [comment]
Complete spelling revision for the ES translation (https://github.com/learnlatex/learnlatex.github.io/commit/a7faa62e9425d324e687284289620457dc74716b) thanks @JhonatanDczel
